### DISCUSSION TITLE
Updating my Annotating script

### DISCUSSION THREAD

**Discussion by Schievelbein** [ImageMagick/ImageMagick#2899] [Digital Image Processing]
Years ago, I developed a script to annotate all of the photos that I scan.  
I would like to update it to V7 and possibly use Magick if needed in order to simplify and stop reading the file 3 times...
Any and all suggestions of shortening the script would be appreciated.

convert $1 \( -clone 0 -size $(identify -format "%w" $1) -delete 0 -background Black -gravity center -pointsize $(identify -format "%[fx:h*.02]" $1 ) -fill White caption:"_ \n $(identify -format "%[IPTC:2:120]" $1 )" -trim \) -append $1_

I was hoping to get rid of the "identify' commands if possible.
Thank you

**Comment by fmw42** [comment]
I think this is what you want for IM 7. In IM 7 you can use % escapes in most functions or use -set option %xxx to create an option that can be called later.

`magick $1 ( -size "%w" -background black -gravity center -pointsize "%[fx:h*.02]" -fill white caption:"_\n %[IPTC:2:120]" -trim ) -delete 0 -append $1_`

Do you actually want the underscore at the end? You may also want to add +repage after the -trim. Also if you delete 0, then you have only one image left and so no two images to append.

**Reply by Schievelbein** [reply]
Thank you for responding.
I tried what you wrote on 3 images and get the following error on each, while not when using the V6 version...
magick: unknown image property "%[IPTC:2:120]" @ warning/property.c/InterpretImageProperties/4111.
I tried using the Identify portion and it also gives the error.
There must have been a change in how to retireve the Caption/Description filed on photos in the upgrade from V6 (convert) to V7 Magick ???  Here is the test file I use and the result from my script....
![Test](https://user-images.githubusercontent.com/7518486/100795802-e0118b80-33dc-11eb-9f46-f2aa0730808c.jpg)
![Test_result](https://user-images.githubusercontent.com/7518486/100795861-f7507900-33dc-11eb-837f-518a7abba6a9.jpg)

**Reply by fmw42** [reply]
We need to use -set option to declare the variables. So this works for me:

magick img.jpg -set option:ww "%w" -set option:iptc "%[IPTC:2:120]" -set option:psize "%[fx:h*0.02]" ( -size "%[ww]" -background black -gravity center -pointsize "%[psize]" -fill white caption:"_\n %[iptc]" -trim +repage) -append result.jpg

![result](https://user-images.githubusercontent.com/13458218/100798111-04b73480-33d8-11eb-885e-0a0900261d1d.jpg)

**Reply by Schievelbein** [reply]
Thank you for your help.  This works great from the command line, I just need to figure out in Linux Mint how to attach it to files to "Open With", as this one does not work like my old one did.  THANKS.

**Comment by fmw42** [comment]
I do not understand your "Open With"  What does that mean?  In Linux, the parentheses need to be escaped as \( and \). If you need a method to process when the file is uploaded, then perhaps see the Linux tool, cron.

**Reply by Schievelbein** [reply]
Just for info, I am running Linux Mint 20 and previously, I had set up so that when I right clicked on a tiff or jpg, that I could run the Title script against that file, or sun it from the command line if needed.  When I take the script you wrote and put it into the Title file, set it to execute and try it from the command line, it works great.  If I try to do it using the shortcut, it fails and I cannot figure out why, so I am asking in the Mint forums for ideas.  Thank you for your help.
![Screenshot from 2020-12-01 15-14-46](https://user-images.githubusercontent.com/7518486/100803424-5ebff600-33e8-11eb-8492-de0585e63c36.png)
